sklearn/tree/tests/test_tree.py

```diff
@@ -2617,7 +2617,25 @@ def test_deterministic_pickle():
     assert pickle1 == pickle2
 
 
-def test_regression_tree_missing_values_toy():
+@pytest.mark.parametrize(
+    "X",
+    [
+        np.array([np.nan, 2, np.nan, 4, 5, 6]).reshape(
+            -1, 1
+        ),  # missing values will go left for greedy splits
+        np.array([np.nan, np.nan, 3, 4, 5, 6]).reshape(
+            -1, 1
+        ),  # missing values will go left for greedy splits
+        np.array([1, 2, 3, 4, np.nan, np.nan]).reshape(
+            -1, 1
+        ),  # missing values will go right for greedy splits
+        np.array([1, 2, 3, 4, np.nan, 6]).reshape(
+            -1, 1
+        ),  # missing values will go right for greedy splits
+    ],
+)
```

Comment:
We can make this more compact if we add a line `X = X.reshape(-1, 1)` in the test below
```suggestion
@pytest.mark.parametrize(
    "X",
    [
        # missing values will go left for greedy splits
        np.array([np.nan, 2, np.nan, 4, 5, 6]),
        np.array([np.nan, np.nan, 3, 4, 5, 6]),
        # missing values will go right for greedy splits
        np.array([1, 2, 3, 4, np.nan, np.nan]),
        np.array([1, 2, 3, np.nan, 6, np.nan]),
    ],
)
```